File: test/cpp/test_delete_event_group.cpp
// Your First C++ Program

#include <iostream>
#include <time.h>
#include "../src/harmonie_reader.h"

using namespace Harmonie;

int main() {
    HarmonieReader reader;
    std::cout << "Test: test_write_events" << std::endl;
    std::string filename = "";
    std::string channel = "C3-A2";
    int montageIndex = 0;
    if (filename == "") {
        std::cout << "ERROR: No filename specified" << std::endl;
        return 0;
    }

    std::cout << "Opening file... :" << filename << std::endl;
    bool res = reader.openFile(filename);

    if (!res) {
        std::cout << "ERROR Failed to open the file:" << filename << std::endl;
        return 0;
    } else {
        std::cout << "SUCCESS" << std::endl;
    }

    // Adding test group
    std::string groupName = "TestGroup";
    std::string groupDescription = "TestGroup Description";
    std::cout << "Adding event group... " << std::endl;
    std::cout << "  Group name: " << groupName << std::endl;
    std::cout << "  Description:" << groupDescription << std::endl << std::endl;

    // Create the test event
    
    std::string eventName = "test_event";
    std::string eventDesc = "test_event_description";

    double startTime = 7;
    double timeLength = 1.5;
    
    for (int i=0; i < 10; i++) {
        std::cout << "Adding event:\"" << eventName << "\" Start: " << startTime << " Length: " << timeLength <<std::endl;
        startTime += 6;
        std::vector<std::string> channels;
        channels.push_back(channel);

        // Add it 
        reader.addEvent(eventName, groupName, startTime, timeLength, channels, montageIndex);
    }

    // Save file
    std::cout << "Saving file..." << std::endl;
    reader.saveFile();
    std::cout << "Closing file..." << std::endl;
    reader.closeFile();

    // Validate if the event has been created
    HarmonieReader reader_to_delete;

    std::cout << "Reopening file... :" << filename << std::endl;
    res = reader_to_delete.openFile(filename);

    if (!res) {
        std::cout << "ERROR Failed to open the file:" << filename << std::endl;
        return 0;
    } else {
        std::cout << "SUCCESS" << std::endl;
    }

    std::cout << "List of events before deleting:" << std::endl;
    std::vector<CPSGFile::EVITEM> events = reader_to_delete.getEvents();
    
    for (const auto& event: events) {
        std::cout << "  " << event.Name << std::endl;
    }

    std::cout << "Deleting event group: " << groupName << std::endl;
    reader_to_delete.removeEventsByGroup(groupName);

    // Save file
    std::cout << "Saving file..." << std::endl;
    reader_to_delete.saveFile();
    std::cout << "Closing file..." << std::endl << std::endl;
    reader_to_delete.closeFile();

    // Validate if the event has been created
    std::cout << "Checking if the event are really deleted..." << std::endl;
    HarmonieReader validation_reader;
    res = validation_reader.openFile(filename);

    events = validation_reader.getEvents();
    
    std::cout << "List of events after deleting:" << std::endl;
    
    for (const auto& event: events) {
        std::cout << "  \"" << event.Name << "\"" << std::endl;
    }

    for (const auto& event: events) {
        if( event.Name == eventName) {
            std::cout << "FAILED The event was not removed" << std::endl;
            validation_reader.closeFile();
            return 0;
        }
    }
    
    std::cout << "SUCCESS The events were correctly removed." << std::endl;

    std::cout << "Closing file..." << std::endl << std::endl;
    validation_reader.closeFile();

    std::cout << "DONE" << std::endl;
    return 0;
}

File: test/python/test_delete_event_group.py

import time
from sys import platform
import sys
sys.path.insert(0, '')

if platform == "linux" or platform == "linux2":
    import build.HarmonieReader as HarmonieReader
elif platform == "darwin":
    import build.HarmonieReader as HarmonieReader
elif platform == "win32":
    import build.Release.HarmonieReader as HarmonieReader

reader = HarmonieReader.HarmonieReader()
print("Test: test_write_events")

filename = ""
if filename == "":
    print("ERROR: No filename specified")
    quit()

print(f"Opening file... :{filename}")
res = reader.open_file(filename)

# Adding a test group
group_name = "TestGroup"
group_desc = "TestGroup description"

print(f"Adding event group...")
print(f"  Group name:{group_name}")
print(f"  Group desc:{group_desc}")

group_index = reader.add_event_group(group_name, group_desc)

channel = "C3-A2"
event_name = "test_delete_event_group_name_" +  str(int(time.time()))
event_desc = "test_delete_event_group_description"

start_time = 10
length = 1.5
montage_index = 0
number_of_events_to_create = 10

print(f"Adding {number_of_events_to_create} events  name:{event_name}")
channels = []
channels.append(channel)

for i in range(number_of_events_to_create):
    print(f"    event:{event_name} start:{start_time}")
    reader.add_event(event_name, group_name, start_time, length, channels, montage_index)
    start_time = start_time + 5

print("Saving file...")
reader.save_file()
print("Closing file...")
reader.close_file()

print(f"Reopening file...:{filename}")

reader_to_delete = HarmonieReader.HarmonieReader()
reader_to_delete.open_file(filename)

print("List of events before deleting:")
events = reader_to_delete.get_events()

for e in events:
    print(f"    {e.name}")

print(f"Deleting event group:{group_name}")
reader_to_delete.remove_events_by_group(group_name)

print("Saving file...")
reader_to_delete.save_file()

print("Closing file...")
reader_to_delete.close_file()

print("Checking if the events were really deleted...")
validation_reader = HarmonieReader.HarmonieReader()
validation_reader.open_file(filename)
events = validation_reader.get_events()

for e in events:
    print(f"    {e.name}")

for e in events:
    if e.name == event_name:
        print("FAILED The event was not removed")
        validation_reader.close_file()
        quit()

print("SUCCESS The events were correctly removed.")
validation_reader.close_file()
print("DONE")
